src/main.cpp

```diff
@@ -35,15 +35,26 @@ int main(int argc, char* argv[])
     backup_camera->init_graphics(renderer);
     printf("Starting threads\n");
     backup_camera->start_threads();
+    bool setFullscreenNext = true;
 
-    while (backup_camera->process_events())
+    while (int fullscreenToggleCheck = backup_camera->process_events())
     {
         if (backup_camera->update())
         {
             SDL_RenderPresent(renderer);
             SDL_SetRenderDrawColor(renderer, 0, 0, 0, 0);
             SDL_RenderClear(renderer);
         }
+
+        //check for fullscreen toggle
+        if (fullscreenToggleCheck == 2)
```

Comment:
This `2` is what we call a magic number, it should be represented as some sort of variable.